### PR TITLE
Revise Indexing Advisor and Indexing Engine documentation for 3.0 launch

### PR DESCRIPTION
**Pull request by lfittl**
## TODO

* [x] Update diagram that shows phases (since I adjusted index selection to be its own distinct phase from costing, similar to how we treat it now in-app)
* [x] Improve CP model intro page (see TODO inline)
* [x] Preset docs page

### CODE REVIEW

File: index-advisor/getting-started.mdx [hunk 1]
```diff
@@ -79,9 +85,9 @@ your server.
 
 ## How to navigate the insights overview
 
-![Overview of missing index insights](/missing-index-list.png)
+![Overview of missing index insights](/ia-overview-missing-indexes.png)
 
-At the top of the overview page you find KPIs that frame our insight
+At the top of the overview page you find KPIs that frame the insight
```
Comment (msakrejda) [inline index-advisor/getting-started.mdx:90]: I guess it's an existing issue, but "KPI" may be too niche a term to use here.
```suggestion
At the top of the overview page you find relevant metrics that frame the insight
```

File: index-advisor/getting-started.mdx [hunk 2]
```diff
@@ -104,19 +110,19 @@ missing and unused index insights below.
...
-Queries fetch data from tables through different types of scans, and some scans
-can be executed more efficiently if an appropriate index is present. The Index
+Detects tables that have potential to be improved by adding indexes. Queries
+fetch data from tables through different types of scans, and some scans
+can be executed more efficiently if appropriate indexes are present. The Index
 Advisor attempts to infer what indexes would be helpful for your query workload.
 See more details in the [Indexing Engine documentation](/docs/indexing-engine).
 
-The impact of a missing index insight is measured by cost improvement.
+The impact of a missing index insight is measured by weighted cost improvement.
```
Comment (msakrejda) [inline index-advisor/getting-started.mdx:125]: Should we explain what that is?

**Review by jensy — APPROVED** [review]
#### Index Advisor high-level docs
Nothing stands out for me that needs revision

#### Indexing Engine
- Duplication of heading on the Index Write Overhead page (this is already the case today)

File: indexing-engine/cp-model/index.mdx [hunk 3]
```diff
@@ -1,19 +1,58 @@
...
+>
+</iframe>
+<br /><br />
+
+## Built-in configurations
+
+[Depending on the workload](/docs/index-advisor/configurations), the Indexing Engine will automatically
+default to one of the available built-in model configurations:
+
+- [Write-optimized](cp-model/built-in-configs#write-optimized) (less indexes, higher scan costs)
```
Comment (msakrejda) [inline indexing-engine/cp-model/index.mdx:37]:
```suggestion
- [Write-optimized](cp-model/built-in-configs#write-optimized) (fewer indexes, higher scan costs)
```

**Review by msakrejda — APPROVED** [review]

File: indexing-engine/what-if-analysis.mdx [hunk 4]
```diff
@@ -51,5 +51,5 @@ You can read more behind the scenes the details in our blog post,
 [How we deconstructed the Postgres planner to find indexing insights](/blog/deconstructing-the-postgres-planner).
 
 After "What if?" analysis has completed, we utilize the
-["Good Enough" Index Selection Algorithm](/docs/indexing-engine/good-enough-index-selection)
+[Index Selection](/docs/indexing-engine/good-enough-index-selection) model
```
Comment (keiko713) [inline indexing-engine/what-if-analysis.mdx:54]: bad link? /docs/indexing-engine/good-enough-index-selection is gone

Comment (keiko713) [reply indexing-engine/what-if-analysis.mdx:54]: Well, technically there is a redirect in https://github.com/pganalyze/pganalyze-web/pull/509 so I guess it's fine as is

**Review by keiko713 — APPROVED** [review]
Looking good!

Can be done as a follow-up PR but:
* https://pganalyze-we-indexing-e-5efknj.herokuapp.com/docs/index-advisor/checks-and-alerts & TOC of Alerts & Check-Up
  * update https://pganalyze.com/docs/checks/index_advisor/missing_index to https://pganalyze.com/docs/checks/index_advisor/indexing_engine
* https://pganalyze-we-indexing-e-5efknj.herokuapp.com/docs/index-advisor
  * update this to include "Experimenting With Different Configurations"

**Pull request merged**